### PR TITLE
[IMP] rating: Introduce new code

### PR DESCRIPTION
**Pull request by maki-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/core_ui/message.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
                             <t t-if="message.trackingValues.length">
                                 <t t-call="mail.tracking_values"/>
                             </t>
-                            <div t-if="message.body" class="o-mail-message-body text-break mb-0 bg-view">
+                            <div t-if="message.body" class="o-mail-message-body text-break mb-0 bg-view w-100">
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core_ui/message.xml:85]: strange to have to add width instruction on a child that has a flex parent, what does it fix?

File: addons/mail/static/src/new/web/messaging_menu/messaging_menu.xml [hunk 2]
```diff
@@ -79,7 +79,9 @@
                             <i class="fa fa-mail-reply me-1"/>You:
                         </t>
                         <t t-else="">
-                            <t t-esc="preview.mostRecentMsg.author?.name"/>:
+                            <t t-if="preview.mostRecentMsg.author">
+                                <t t-esc="preview.mostRecentMsg.author.name"/>:
+                            </t>
```
Comment (seb-odoo) [inline addons/mail/static/src/new/web/messaging_menu/messaging_menu.xml]: What was happening with the `t-esc` if the author was undefined before this change?

Comment (seb-odoo) [reply addons/mail/static/src/new/web/messaging_menu/messaging_menu.xml]: Oh I see it's for the `:`, but then I should suggest to have `else + if` `elif` instead

File: addons/rating/static/src/new/messaging_menu.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.messaging_menu.content" t-inherit-mode="extension">
+        <xpath expr="//NotificationItem[@body='preview.body']" position="attributes">
+            <attribute name="rating">preview.mostRecentMsg?.rating</attribute>
```
Comment (seb-odoo) [inline addons/rating/static/src/new/messaging_menu.xml:5]: Shouldn't the same possibly also apply to `mostRecentNeedactionMsg`?

File: addons/rating/static/src/new/notification_item.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.notification_item" t-inherit-mode="extension">
+        <xpath expr="//*[@name='notificationBody']" position="replace">
+            <t t-if="props.rating">
+                <span>Rating:</span>
+                <img class="o-rating-preview-image ms-2" t-att-src="props.rating.ratingImageUrl" t-att-alt="props.rating.ratingText"/>
+        </t>
+        <t t-else="">$0</t>
+            </xpath>
```
Comment (seb-odoo) [inline addons/rating/static/src/new/notification_item.xml]: indentation issue in this file

File: addons/test_mail_full/static/tests/qunit_suite_tests/channel_preview_view_tests.js [hunk 5]
```diff
@@ -1,55 +1,33 @@
...
+import { start, startServer } from "@mail/../tests/helpers/test_utils";
+import { getFixture, triggerEvent } from "@web/../tests/helpers/utils";
 
-QUnit.module('test_mail_full', {}, function () {
-QUnit.module('channel_preview_view_tests.js');
+let target;
 
-QUnit.skipRefactoring('rating value displayed on the thread preview', async function (assert) {
-    assert.expect(4);
+QUnit.module("test_mail_full", {
```
Comment (seb-odoo) [inline addons/test_mail_full/static/tests/qunit_suite_tests/channel_preview_view_tests.js]: The odoo module name `test_mail_full` is automatically added now, a better qunit module name would be the name of the test file.

File: addons/test_mail_full/static/tests/qunit_suite_tests/channel_preview_view_tests.js [hunk 6]
```diff
@@ -1,55 +1,33 @@
...
-        "top",
-        "should contain the correct rating text"
-    );
-});
-
+    await start();
+    await triggerEvent(target, ".o_menu_systray i[aria-label='Messages']", "click");
+    assert.containsOnce(target, ".o-mail-notification-item-inlineText:contains(Rating:)");
+    assert.containsOnce(target, "img[data-alt='top']");
+    assert.containsOnce(target, "img[data-src='/rating/static/src/img/rating_5.png']");
```
Comment (seb-odoo) [inline addons/test_mail_full/static/tests/qunit_suite_tests/channel_preview_view_tests.js]: these assert are a bit too generic, it's not clear which image is being targeted

**Review by seb-odoo — APPROVED** [review]
Ok except comments

File: addons/mail/static/src/new/core_ui/message.xml (hunk 1 above)
Comment (maki-odoo) [reply addons/mail/static/src/new/core_ui/message.xml:85]: Without width, It is like this:

Comment (maki-odoo) [reply addons/mail/static/src/new/core_ui/message.xml:85]: Should be:

File: addons/rating/static/src/new/messaging_menu.xml (hunk 3 above)
Comment (maki-odoo) [reply addons/rating/static/src/new/messaging_menu.xml:5]: The `channel preview` and `thread needaction preview` are same component in the new code unlike the old code. The `mostRecentMsg` and `mostRecentNeedactionMsg` are same message in this part of code.  I am not sure If it is the point of comment although.

File: addons/mail/static/src/new/web/messaging_menu/messaging_menu.xml (hunk 2 above)
Comment (maki-odoo) [reply addons/mail/static/src/new/web/messaging_menu/messaging_menu.xml:84]: In messages without  author (for example nonusers in live chat) the `:` symbol remains without author name. Actually if there is not author name we don't want `:` either.

File: addons/rating/static/src/new/messaging_menu.xml (hunk 3 above)
Comment (maki-odoo) [reply addons/rating/static/src/new/messaging_menu.xml:5]: I added `thread needaction preview` test to this PR.

Comment (seb-odoo) [reply addons/rating/static/src/new/messaging_menu.xml:5]: Depending on the preview type it's not the same message that is displayed, needaction does not display the `mostRecentMsg`, it displays the `mostRecentNeedactionMsg`.

**Comment by seb-odoo** [comment]
Runbot is red, could you rebase to fix (some of) the issues?

**Comment by maki-odoo** [comment]
> Runbot is red, could you rebase to fix (some of) the issues?

Done

File: addons/rating/static/src/new/messaging_menu.xml (hunk 3 above)
Comment (maki-odoo) [reply addons/rating/static/src/new/messaging_menu.xml:5]: Follow up PR for proper use of  `mostRecentMsg` and `mostRecentNeedactionMsg`.

**Review by seb-odoo — APPROVED** [review]

**Pull request merged**